main.cpp

```diff
@@ -286,17 +192,22 @@ struct FloatType
     FloatType &subtract(float rhs);
     FloatType &multiply(float rhs);
     FloatType &divide(float rhs);
+    FloatType &pow(float exp);
+    FloatType &pow(const FloatType &exp);
+    FloatType &pow(const DoubleType &exp);
+    FloatType &pow(const IntType &exp);
```

Comment:
Don't put `&` next to the function name. 
put it next to the return type.
It looks like you're taking the address of the function, instead of returning a reference.
- [ ] revise